File: lab3/task_5/cpp/classes/Jewelry.cpp
#include <iostream>
#ifndef JEWELRY_H
#define JEWELRY_H
using namespace std;

class Jewelry {
  private: 
    double weight =1;
    double pricePerGramm =1;

  public:
    void init(double arg_weight, double arg_pricePerGramm);

    void read();

    void display();

    double getFullPricePerGramm();
};

  void Jewelry::init(double arg_weight, double arg_pricePerGramm) {
    weight = arg_weight;
    pricePerGramm = arg_pricePerGramm;
  }

  void Jewelry::read() {
    cout << "Введите вес в граммах: ";
    cin >> weight;
    cout << "Введите цену за грамм: ";
    cin >> pricePerGramm;
  }

  void Jewelry::display() {
    cout << "Вес: " << weight << endl;
    cout << "Цена за грамм: " << pricePerGramm << endl;
  }

  double Jewelry::getFullPricePerGramm() {
    return weight * pricePerGramm;
  }

#endif

File: lab3/task_5/cs/Jewelry.cs
using System;

namespace Task_4
{
    public class Jewelry
    {
        private double weight = 1;
        private double pricePerGramm = 1;

        public void Init(double arg_weight, double arg_pricePerGramm)
        {
            weight = arg_weight;
            pricePerGramm = arg_pricePerGramm;
        }

        public void Read()
        {
            Console.Write("Введите вес в граммах: ");
            weight = Convert.ToDouble(Console.ReadLine());
            Console.Write("Введите цену за грамм: ");
            pricePerGramm = Convert.ToDouble(Console.ReadLine());
        }


        public void Display()
        {
            Console.WriteLine($"Вес: {weight}");
            Console.WriteLine($"Цена за грамм: {pricePerGramm}");
        }

        public double GetFullPricePerGramm()
        {
            return weight * pricePerGramm;
        }
    }
}
